azurerm/internal/services/securitycenter/resource_arm_security_center_subscription_pricing.go

```diff
@@ -44,6 +52,20 @@ func resourceArmSecurityCenterSubscriptionPricing() *schema.Resource {
 					string(security.Standard),
 				}, false),
 			},
+			"resource_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"AppServices",
+					"ContainerRegistry",
+					"KeyVaults",
+					"KubernetesService",
+					"SqlServers",
+					"SqlServerVirtualMachines",
+					"StorageAccounts",
+					"VirtualMachines",
+				}, false),
```

Comment:
Is there documentation somewhere that exposes these values? Do we know this is an exhaustive list? (The REST docs only indicate a `string` type with no guidance on allowed values)

Reply:
As discussed, unfortunately, there's no documentation on the values that the resource type can take; I got the list of values using the list endpoint (https://docs.microsoft.com/en-us/rest/api/securitycenter/pricings/list).

The problem seems to be in the Azure API specs, which don't define `PricingName` (resource type) as an enum, see https://github.com/Azure/azure-rest-api-specs/blob/3695f4ef3bf65c253509d2708c31c96b71822613/specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/pricings.json#L223.

Created an issue (https://github.com/Azure/azure-rest-api-specs/issues/11006) so that the enum can be used once it gets resolved.